apps/src/StudioApp.js

```diff
@@ -608,6 +608,61 @@ StudioApp.prototype.init = function (config) {
   }
 };
 
+StudioApp.prototype.hasValidContainedLevelResult_ = function () {
+  var results = this.getContainedLevelResults();
+  if (results.length !== 1) {
+    throw "Exactly one contained level result is currently required.";
+  }
+  var firstResult = results[0];
+  return typeof firstResult.result.response !== 'undefined' &&
+      firstResult.result.response !== '';
+};
+
+StudioApp.prototype.notifyInitialRenderComplete = function (config) {
+  if (config.hasContainedLevels && config.containedLevelOps) {
+    this.lockContainedLevelAnswers = config.containedLevelOps.lockAnswers;
+    this.getContainedLevelResults = config.containedLevelOps.getResults;
+
+    $('#containedLevel0').appendTo($('#containedLevelContainer'));
+
+    if (this.hasValidContainedLevelResult_()) {
+      // We already have an answer, don't allow it to be changed, but allow Run
+      // to be pressed so the code can be run again.
+      this.lockContainedLevelAnswers();
+    } else {
+      // No answers yet, disable Run button until there is an answer
+      $('#runButton').prop('disabled', true);
```

Comment:
Agreed that it's not ideal, but until the `#runButton` is fully React, it seems strange to build a different state-based rendering mechanism.

Regarding running without the runButton.. there are certainly ways to do it such as using the browser debugger, but I can't think of any easy way to do it using our UI. Blocking runButon execution for all apps (at the StudioApp level) could normally be accomplished in `runButtonClickWrapper`, except that applab replaces that function without calling the "base class", so it seems a little messy.